crates/winch/src/compiler.rs

```diff
@@ -105,21 +116,14 @@ impl wasmtime_environ::Compiler for Compiler {
 
         let mut ret = Vec::with_capacity(funcs.len());
         for (i, (sym, func)) in funcs.iter().enumerate() {
-            let func = &func.downcast_ref::<CompiledFunction>().unwrap().0;
-
-            // TODO: Implement copying over this data into the
-            // `ModuleTextBuilder` type. Note that this should probably be
-            // deduplicated with the cranelift implementation in the long run.
-            assert!(func.relocs().is_empty());
-            assert!(func.traps().is_empty());
-            assert!(func.stack_maps().is_empty());
+            let func = &func.downcast_ref::<CompiledFunction>().unwrap();
 
             let (sym, range) = builder.append_func(
                 &sym,
-                func.data(),
-                self.function_alignment(),
-                None,
-                &[],
+                &func.body,
+                func.alignment,
+                func.unwind_info.as_ref(),
+                &func.relocations,
                 |idx| resolve_reloc(i, idx),
             );
```

Comment:
This is an example where ideally we'd "simply" feed in a `&MachBufferFinalized` into this function so you wouldn't have to pass all these extra fields manually. That's at least the rough idea, unsure how it'll play out in practice.